test/integration/good/code-gen/ode/cpp.expected

```diff
@@ -282,7 +282,77 @@ class ode_adjoint_test_model_model final : public model_base_crtp<ode_adjoint_te
   }
   template <bool propto__, bool jacobian__, typename VecR, typename VecI,
             stan::require_vector_like_t<VecR>* = nullptr,
-            stan::require_vector_like_vt<std::is_integral, VecI>* = nullptr>
+            stan::require_vector_like_vt<std::is_integral, VecI>* = nullptr,
+            stan::require_not_st_var<VecR>* = nullptr>
```

Comment:
This is written in a kind of specific way. I'd like this overload to work if we also use a `stan::math::var_value<Eigen::VectorXd>` or `stan::math::var_value<stan::matrix_cl<double>>`. The latter would be nice if the whole model is on the GPU